src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -661,6 +661,23 @@ gl_sframe drawing_classifier::perform_inference(data_iterator* data) const {
   return writer.close();
 }
 
+gl_sframe infer_feature_column_and_prepare_data(gl_sframe &data) {
```

Comment:
Please move free (non-class) helper functions to the top of the file and put them in an anonymous namespace (or otherwise specify only local linkage)

Also, prefer not to mutate arguments. Could this return a new SFrame, without mutating the input?

Reply:
Done.